kiwi/bootloader/install/grub2.py

```diff
@@ -174,114 +169,119 @@ def install(self):
         ):
-            self._mount_device_and_volumes()
-            shim_install = self._get_shim_install_tool_name(
-                self.root_mount.mountpoint
-            )
-            # if shim-install does _not_ exist the fallback mechanism
-            # has applied at the bootloader/config level and we expect
-            # no further tool calls to be required
-            if shim_install:
-                # Before we call shim-install, the grub installer binary is
-                # replaced by a noop. Actually there is no reason for
-                # shim-install to call the grub installer because it should
-                # only setup the system for EFI secure boot which does not
-                # require any bootloader code in the master boot record.
-                # In addition kiwi has called the grub installer right
-                # before
-                self._disable_grub2_install(self.root_mount.mountpoint)
-                Command.run(
-                    [
-                        'chroot', self.root_mount.mountpoint,
-                        'shim-install', '--removable'
-                    ] + self.shim_install_arguments + [
-                        self.device
-                    ]
+            with ExitStack() as stack:
+                self._mount_device_and_volumes(stack)
+                shim_install = self._get_shim_install_tool_name(
+                    self.root_mount.mountpoint
                 )
-                # restore the grub installer noop
-                self._enable_grub2_install(self.root_mount.mountpoint)
+                # if shim-install does _not_ exist the fallback mechanism
+                # has applied at the bootloader/config level and we expect
+                # no further tool calls to be required
+                if shim_install:
+                    # Before we call shim-install, the grub installer binary is
+                    # replaced by a noop. Actually there is no reason for
+                    # shim-install to call the grub installer because it should
+                    # only setup the system for EFI secure boot which does not
+                    # require any bootloader code in the master boot record.
+                    # In addition kiwi has called the grub installer right
+                    # before
+                    self._disable_grub2_install(self.root_mount.mountpoint)
+                    try:
+                        Command.run(
+                            [
+                                'chroot', self.root_mount.mountpoint,
+                                'shim-install', '--removable'
+                            ] + self.shim_install_arguments + [
+                                self.device
+                            ]
+                        )
+                    finally:
+                        # restore the grub installer noop
+                        self._enable_grub2_install(self.root_mount.mountpoint)
```

Comment:
Does it really make sense to always perform the grub2 installer enablement? If an exception is thrown, then the build will be abandoned and the buildroot is wiped

Reply:
but I should move it into the try block

Reply:
yes this is correct but if the shim-install succeeds you have to revert the crap. What's wrong in trying to revert a really bad change to the root such that shim-install works in an image build ? I really dislike this disable/enable code and would like to get rid of it rather sooner than later but as long as it's needed I think we should try hard to revert any of the stupid changes to the root, no ?

Reply:
@dcermak I'd like to give this change and the other PR that already got merged a test with the integration tests. What's your take on this one, can we continue ? Thanks

Reply:
Sorry, forgot about this :see_no_evil: 

So tbh, I find it a bit odd to always perform this cleanup, even if a failure will be catastrophic and won't have any impact on the host. But it doesn't hurt either. If you think it's better to do it, then feel free to merge. It's not like it's bad, I just find it peculiar and it confuses me.

Reply:
ok, I'd like to keep it. The reason is I think temporary changes done to the system and only needed because of the weakness of a tool should be reverted in any case because I don't want to leave the root system in this state even if the tool call has failed and the build process exited. imho it's strange and confusing that we need this extra code to let shim-install work in an appliance build environment. It is yet another one of these tools that thinks it is only called on target and nowhere else... a wrong assumption which also applies to grub and makes all this crap code necessary. But as I said I don't want to leave the build process with that "crap changes" active in any case, mostly because I don't want people to work with it after the build has failed.